src/services/discordBotServices.ts

```diff
@@ -0,0 +1,27 @@
+import config from '../config/config';
+import { DiscordRoleUpdatedList, DiscordUserRole, env } from '../types/global.types';
+import { generateDiscordBotJwt } from '../utils/generateJwt';
+
+export const updateUserRoles = async (env: env, payload: DiscordUserRole[]): Promise<DiscordRoleUpdatedList> => {
```

Comment:
Why is this being handled as a part of cron job repo? Don't you think this can be done in RDS backend code itself?
Essentially the cron jobs repo should only contain code that contains the scheduler and the task that is invoked.

Maybe I am missing something here.

Reply:
If a http request takes a long time, server can choose to timeout that request with an error. 
- Currently discord allows us to update roles at an average rate of 1/sec. This means our request can be timed out while the RDS backend is updating a large number of roles at this rate. 
- Since Cloudflare docs mention that it does not have any hard limits on the duration for which cron job can run, I've moved this here. 

https://developers.cloudflare.com/workers/platform/limits/

Reply:
Okay, but this repo then no longer remains just "cron-jobs". We might have to find a solution for this.

Reply:
Maybe create an issue/ RFC that would tackle this?